### PR TITLE
maybe_not_update_payment_method() returns unchanged value of $update if payment method is not Amazon Pay.

### LINKED ISSUE #103: Callback for filter woocommerce_subscriptions_update_payment_via_pay_shortcode
**Describe the bug**
Callback for filter `woocommerce_subscriptions_update_payment_via_pay_shortcode` [here](https://github.com/woocommerce/woocommerce-gateway-amazon-pay/blob/1d849cf0385dad43c22b0b9fce9206aac3d16867/includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php#L58) is [this function](https://github.com/woocommerce/woocommerce-gateway-amazon-pay/blob/1d849cf0385dad43c22b0b9fce9206aac3d16867/includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php#L712-L718).

That function should:
1. Return boolean, but return id as is also fine.
2. Return `$update` if it doesn't involve Amazon Pay payment method.

**To Reproduce**
1. Install another plugin that hook to `woocommerce_subscriptions_update_payment_via_pay_shortcode` filter, eg: [eWAY](https://github.com/woocommerce/woocommerce-gateway-eway).
2. Purchase subscriptions products.
3. Change payment of that subscriptions to payment method of other plugin.
4. Step through and not the return value of other plugin's callback to `woocommerce_subscriptions_update_payment_via_pay_shortcode` filter.
5. Notice, this plugin's callback will 'override' whatever value the other plugin returns.

**Expected behavior**
When there is payment method change and it doesn't change to Amazon Pay, the callback of `woocommerce_subscriptions_update_payment_via_pay_shortcode` filter should return `$update`.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only the extension is active.
- [x] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by shendy-a8c**
Fixes #103

### All Submissions:

* [ ] Does your code follow the [Extendables](P75Izm-1JH-p2) standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changes proposed in this Pull Request:

Closes # .

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by msaggiorato — APPROVED** [review]
Good catch. Approved for merge @vbelolapotkov .

**Comment by vbelolapotkov** [comment]
All credits to @shendy-a8c as well as the honor to push merge button 😄

**Comment by shendy-a8c** [comment]
I just sync'd with base branch and added changelog entry.

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 *** Amazon Pay Changelog ***
 
+= 2.0.4 - 2021-xx-xx =
+* Fix - Interference when subcsriptions' payment changes to other payment method.
```
Comment (vbelolapotkov) [inline changelog.txt:4]: Not sure but probably one word is missing:
```suggestion
* Fix - Interference when subcsriptions' payment method changes to other payment method.
```

**Review by vbelolapotkov — APPROVED** [review]

File: changelog.txt (hunk 1 above)
Comment (shendy-a8c) [reply changelog.txt:4]: Didn't think of that but nice addition. Thanks!

Comment (shendy-a8c) [reply changelog.txt]: Updated readme as well 460b9ca.

**Pull request merged**